src/main/java/com/github/wovnio/wovnjava/UrlContext.java

```diff
@@ -0,0 +1,38 @@
+package com.github.wovnio.wovnjava;
+
+import java.net.URL;
+import java.net.MalformedURLException;
+
+class UrlContext {
+    private final URL context;
+
+    UrlContext(URL currentLocation) {
+        this.context = currentLocation;
+    }
+
+    public URL resolve(String location) {
+        try {
+            URL url = new URL(this.context, location);
+            return this.stripSlashDotDot(url);
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    public boolean isSameHost(URL url) {
+        return this.context.getHost().equalsIgnoreCase(url.getHost());
+    }
+
+    /*
+     * Remove sections of the URL path that starts with "/.."
+     */
+    private URL stripSlashDotDot(URL url) throws MalformedURLException {
```

Comment:
I want to add test case for location which start with `/..` and `/../..` .

Reply:
I added a test case for absolute path starting with `/../..`.
I believe the test cases now cover relative path navigation, as well as correctly stripping multiple `/..` at the beginning of a path.